### PR TITLE
Feature/196 single region

### LINKED ISSUE #196: Feature: single NZ crustal region option for MFD targets
A genuine single MFD region configuration with the single inversion constraint budget

 - [x] keep ability to do dual region config
 - [x] add 1 region setup alternative
 - [x] do not affect slip regionalisation

### PR DESCRIPTION
**Pull request by voj**
closes #196

### CODE REVIEW

**Review by chrisbc — APPROVED** [review]

File: src/main/java/nz/cri/gns/NZSHM22/opensha/inversion/NZSHM22_CrustalInversionRunner.java [hunk 1]
```diff
@@ -40,13 +41,25 @@
...
     public NZSHM22_CrustalInversionRunner() {
         super();
     }
 
+    /**
+     * Determines whether NZ is a single region (false) or is split up in TVZ and Sans TVZ (true)
+     * @param enableTvz
+     * @return
+     */
+    public NZSHM22_CrustalInversionRunner setEnableTvz(boolean enableTvz){
```
Comment (chrisbc) [inline src/main/java/nz/cri/gns/NZSHM22/opensha/inversion/NZSHM22_CrustalInversionRunner.java:58]: I suggest we say this is an MFD  only setting - eg `setEnableTvzMFDs`

**Comment by voj** [comment]
avoid adding MFDs for empty region

File: src/main/java/nz/cri/gns/NZSHM22/opensha/inversion/MFDManipulation.java [hunk 2]
```diff
@@ -79,7 +79,7 @@ public static UncertainIncrMagFreqDist addMfdUncertainty(IncrementalMagFreqDist
         EvenlyDiscretizedFunc stdDevs = new EvenlyDiscretizedFunc(mfd.getMinX(), mfd.getMaxX(), mfd.size());
         for (int i = 0; i < stdDevs.size(); i++) {
             double rate = mfd.getY(i);
-            double stdDev = ((i < minMagBin) || (maxMagBin < i))? 1e-20 : firstWeightPower / Math.pow(rate, power - 1);
+            double stdDev = ((i < minMagBin) || (maxMagBin < i) || rate == 1e-20 )? 1e-20 : firstWeightPower / Math.pow(rate, power - 1);
```
Comment (chrisbc) [inline src/main/java/nz/cri/gns/NZSHM22/opensha/inversion/MFDManipulation.java:82]: Add TODO to remove 1e-20 when we're happy

File: src/main/java/nz/cri/gns/NZSHM22/opensha/inversion/NZSHM22_CrustalInversionRunner.java (hunk 1 above)
Comment (voj) [reply src/main/java/nz/cri/gns/NZSHM22/opensha/inversion/NZSHM22_CrustalInversionRunner.java:58]: done

File: src/main/java/nz/cri/gns/NZSHM22/opensha/inversion/MFDManipulation.java (hunk 2 above)
Comment (voj) [reply src/main/java/nz/cri/gns/NZSHM22/opensha/inversion/MFDManipulation.java:82]: done

**Pull request closed without merging**